pkg/util/util_test.go

```diff
@@ -0,0 +1,42 @@
+package util
+
+import (
+	"testing"
+	"time"
+)
+
+const (
+	testAttempts = 1
+)
+
+func TestValidateFetchReadURL(t *testing.T) {
+	tests := []struct {
+		url     string
+		succeed bool
+	}{
+		{"http://example.com", true},
+		{"invalid.url.^&*!@#", false},
+	}
+
+	for _, tt := range tests {
+		_, err := ValidateFetchReadURL(tt.url, testAttempts)
+		if err != nil && tt.succeed {
+			t.Errorf("ValidateFetchReadURL was expected to succeed for the url: %v, but it failed with the error: %v", tt.url, err)
+		}
+	}
+}
+
+func TestFetchURLWithRetries(t *testing.T) {
+	test := struct {
+		url     string
+		succeed bool
+	}{
+		"example.com", true,
```

Comment:
I'd like to see here more cases or you can maybe cover it well even from the TestValidateFetchReadURL. But I am generally missing a fail case here and an empty string case.